File: android/src/main/java/com/supermap/rnsupermap/JSFacilityAnalystSetting.java
/**
 * Created by Yangshanglong on 2018/7/11.
 */
package com.supermap.rnsupermap;

import com.facebook.react.bridge.Arguments;
import com.facebook.react.bridge.Promise;
import com.facebook.react.bridge.ReactApplicationContext;
import com.facebook.react.bridge.ReactContextBaseJavaModule;
import com.facebook.react.bridge.ReactMethod;
import com.facebook.react.bridge.WritableArray;
import com.facebook.react.bridge.WritableMap;
import com.supermap.analyst.networkanalyst.FacilityAnalystSetting;
import com.supermap.analyst.networkanalyst.FacilityAnalystSetting;
import com.supermap.analyst.networkanalyst.WeightFieldInfo;
import com.supermap.analyst.networkanalyst.WeightFieldInfos;
import com.supermap.data.DatasetVector;

import java.util.Calendar;
import java.util.HashMap;
import java.util.Map;

public class JSFacilityAnalystSetting extends ReactContextBaseJavaModule {
    public static final String REACT_CLASS = "JSFacilityAnalystSetting";
    public static Map<String , FacilityAnalystSetting> m_FacilityAnalystSettingList = new HashMap<String , FacilityAnalystSetting>();
    FacilityAnalystSetting m_FacilityAnalystSetting;

    public JSFacilityAnalystSetting(ReactApplicationContext context){
        super(context);
    }

    public static FacilityAnalystSetting getObjFromList(String id) {
        return m_FacilityAnalystSettingList.get(id);
    }

    @Override
    public String getName(){return REACT_CLASS;}

    public static String registerId(FacilityAnalystSetting FacilityAnalystSetting){
        for(Map.Entry entry:m_FacilityAnalystSettingList.entrySet())
        {
            if(FacilityAnalystSetting.equals(entry.getValue())){
                return (String) entry.getKey();
            }
        }

        Calendar calendar = Calendar.getInstance();
        String id=Long.toString(calendar.getTimeInMillis());
        m_FacilityAnalystSettingList.put(id,FacilityAnalystSetting);
        return id;
    }

    @ReactMethod
    public void createObj(Promise promise){
        try{
            FacilityAnalystSetting facilityAnalystSetting = new FacilityAnalystSetting();
            String facilityAnalystSettingId = registerId(facilityAnalystSetting);

            promise.resolve(facilityAnalystSettingId);
        }catch (Exception e){
            promise.reject(e);
        }
    }

    /**
     * 返回障碍弧段的 ID 列表
     * @param facilityAnalystSettingId
     * @param promise
     */
    @ReactMethod
    public void getBarrierEdges(String facilityAnalystSettingId, Promise promise) {
        try{
            FacilityAnalystSetting facilityAnalystSetting = getObjFromList(facilityAnalystSettingId);
            int[] arr = facilityAnalystSetting.getBarrierEdges();

            WritableArray wArr = Arguments.fromArray(arr);

            promise.resolve(wArr);
        }catch (Exception e){
            promise.reject(e);
        }
    }

    /**
     * 返回障碍结点的 ID 列表
     * @param facilityAnalystSettingId
     * @param promise
     */
    @ReactMethod
    public void getBarrierNodes(String facilityAnalystSettingId, Promise promise) {
        try{
            FacilityAnalystSetting facilityAnalystSetting = getObjFromList(facilityAnalystSettingId);
            int[] arr = facilityAnalystSetting.getBarrierNodes();

            WritableArray wArr = Arguments.fromArray(arr);

            promise.resolve(wArr);
        }catch (Exception e){
            promise.reject(e);
        }
    }

    /**
     * 返回流向字段
     * @param facilityAnalystSettingId
     * @param promise
     */
    @ReactMethod
    public void getDirectionField(String facilityAnalystSettingId, Promise promise) {
        try{
            FacilityAnalystSetting facilityAnalystSetting = getObjFromList(facilityAnalystSettingId);
            String directionField = facilityAnalystSetting.getDirectionField();

            promise.resolve(directionField);
        }catch (Exception e){
            promise.reject(e);
        }
    }

    /**
     * 返回网络数据集中标识弧段 ID 的字段
     * @param facilityAnalystSettingId
     * @param promise
     */
    @ReactMethod
    public void getEdgeIDField(String facilityAnalystSettingId, Promise promise) {
        try{
            FacilityAnalystSetting facilityAnalystSetting = getObjFromList(facilityAnalystSettingId);
            String edgeIDField = facilityAnalystSetting.getEdgeIDField();

            promise.resolve(edgeIDField);
        }catch (Exception e){
            promise.reject(e);
        }
    }

    /**
     * 返回网络数据集中标识弧段起始结点 ID 的字段
     * @param facilityAnalystSettingId
     * @param promise
     */
    @ReactMethod
    public void getFNodeIDField(String facilityAnalystSettingId, Promise promise) {
        try{
            FacilityAnalystSetting facilityAnalystSetting = getObjFromList(facilityAnalystSettingId);
            String fNodeIDField = facilityAnalystSetting.getFNodeIDField();

            promise.resolve(fNodeIDField);
        }catch (Exception e){
            promise.reject(e);
        }
    }

    /**
     * 返回网络数据集
     * @param facilityAnalystSettingId
     * @param promise
     */
    @ReactMethod
    public void getNetworkDataset(String facilityAnalystSettingId, Promise promise) {
        try{
            FacilityAnalystSetting facilityAnalystSetting = getObjFromList(facilityAnalystSettingId);
            DatasetVector datasetVector = facilityAnalystSetting.getNetworkDataset();
            String datasetVectorId = JSDatasetVector.registerId(datasetVector);

            promise.resolve(datasetVectorId);
        }catch (Exception e){
            promise.reject(e);
        }
    }

    /**
     * 返回网络数据集中标识网络结点 ID 的字段
     * @param facilityAnalystSettingId
     * @param promise
     */
    @ReactMethod
    public void getNodeIDField(String facilityAnalystSettingId, Promise promise) {
        try{
            FacilityAnalystSetting facilityAnalystSetting = getObjFromList(facilityAnalystSettingId);
            String nodeIDField = facilityAnalystSetting.getNodeIDField();

            promise.resolve(nodeIDField);
        }catch (Exception e){
            promise.reject(e);
        }
    }

    /**
     * 返回网络数据集中标识弧段终止结点 ID 的字段
     * @param facilityAnalystSettingId
     * @param promise
     */
    @ReactMethod
    public void getTNodeIDField(String facilityAnalystSettingId, Promise promise) {
        try{
            FacilityAnalystSetting facilityAnalystSetting = getObjFromList(facilityAnalystSettingId);
            String tNodeIDField = facilityAnalystSetting.getTNodeIDField();

            promise.resolve(tNodeIDField);
        }catch (Exception e){
            promise.reject(e);
        }
    }

    /**
     * 返回点到弧段的距离容限
     * @param facilityAnalystSettingId
     * @param promise
     */
    @ReactMethod
    public void getTolerance(String facilityAnalystSettingId, Promise promise) {
        try{
            FacilityAnalystSetting facilityAnalystSetting = getObjFromList(facilityAnalystSettingId);
            Double tolerance = facilityAnalystSetting.getTolerance();

            promise.resolve(tolerance);
        }catch (Exception e){
            promise.reject(e);
        }
    }

    /**
     * 返回权值字段信息集合对象
     * @param facilityAnalystSettingId
     * @param promise
     */
    @ReactMethod
    public void getWeightFieldInfos(String facilityAnalystSettingId, Promise promise) {
        try{
            FacilityAnalystSetting facilityAnalystSetting = getObjFromList(facilityAnalystSettingId);
            WeightFieldInfos weightFieldInfos = facilityAnalystSetting.getWeightFieldInfos();

            String weightFieldInfosId = JSWeightFieldInfos.registerId(weightFieldInfos);

            promise.resolve(weightFieldInfosId);
        }catch (Exception e){
            promise.reject(e);
        }
    }

    /**
     * 设置障碍弧段的 ID 列表
     * @param facilityAnalystSettingId
     * @param value
     * @param promise
     */
    @ReactMethod
    public void setBarrierEdges(String facilityAnalystSettingId, int[] value, Promise promise) {
        try{
            FacilityAnalystSetting facilityAnalystSetting = getObjFromList(facilityAnalystSettingId);
            facilityAnalystSetting.setBarrierEdges(value);

            promise.resolve(true);
        }catch (Exception e){
            promise.reject(e);
        }
    }

    /**
     * 设置障碍结点的 ID 列表
     * @param facilityAnalystSettingId
     * @param value
     * @param promise
     */
    @ReactMethod
    public void setBarrierNodes(String facilityAnalystSettingId, int[] value, Promise promise) {
        try{
            FacilityAnalystSetting facilityAnalystSetting = getObjFromList(facilityAnalystSettingId);
            facilityAnalystSetting.setBarrierNodes(value);

            promise.resolve(true);
        }catch (Exception e){
            promise.reject(e);
        }
    }

    /**
     * 设置流向字段
     * @param facilityAnalystSettingId
     * @param value
     * @param promise
     */
    @ReactMethod
    public void setDirectionField(String facilityAnalystSettingId, String value, Promise promise) {
        try{
            FacilityAnalystSetting facilityAnalystSetting = getObjFromList(facilityAnalystSettingId);
            facilityAnalystSetting.setDirectionField(value);

            promise.resolve(true);
        }catch (Exception e){
            promise.reject(e);
        }
    }

    /**
     * 设置网络数据集中标识弧段 ID 的字段
     * @param facilityAnalystSettingId
     * @param value
     * @param promise
     */
    @ReactMethod
    public void setEdgeIDField(String facilityAnalystSettingId, String value, Promise promise) {
        try{
            FacilityAnalystSetting facilityAnalystSetting = getObjFromList(facilityAnalystSettingId);
            facilityAnalystSetting.setEdgeIDField(value);

            promise.resolve(true);
        }catch (Exception e){
            promise.reject(e);
        }
    }

    /**
     * 设置网络数据集中标识弧段起始结点 ID 的字段
     * @param facilityAnalystSettingId
     * @param promise
     */
    @ReactMethod
    public void setFNodeIDField(String facilityAnalystSettingId, String value, Promise promise) {
        try{
            FacilityAnalystSetting facilityAnalystSetting = getObjFromList(facilityAnalystSettingId);
            facilityAnalystSetting.setFNodeIDField(value);

            promise.resolve(true);
        }catch (Exception e){
            promise.reject(e);
        }
    }

    /**
     * 设置网络数据集
     * @param facilityAnalystSettingId
     * @param datasetVectorId
     * @param promise
     */
    @ReactMethod
    public void setNetworkDataset(String facilityAnalystSettingId, String datasetVectorId, Promise promise) {
        try{
            FacilityAnalystSetting facilityAnalystSetting = getObjFromList(facilityAnalystSettingId);
            DatasetVector datasetVector = JSDatasetVector.getObjFromList(datasetVectorId);
            facilityAnalystSetting.setNetworkDataset(datasetVector);

            promise.resolve(true);
        }catch (Exception e){
            promise.reject(e);
        }
    }

    /**
     * 设置网络数据集中标识网络结点 ID 的字段
     * @param facilityAnalystSettingId
     * @param value
     * @param promise
     */
    @ReactMethod
    public void setNodeIDField(String facilityAnalystSettingId, String value, Promise promise) {
        try{
            FacilityAnalystSetting facilityAnalystSetting = getObjFromList(facilityAnalystSettingId);
            facilityAnalystSetting.setNodeIDField(value);

            promise.resolve(true);
        }catch (Exception e){
            promise.reject(e);
        }
    }

    /**
     * 设置网络数据集中标识弧段终止结点 ID 的字段
     * @param facilityAnalystSettingId
     * @param value
     * @param promise
     */
    @ReactMethod
    public void setTNodeIDField(String facilityAnalystSettingId, String value, Promise promise) {
        try{
            FacilityAnalystSetting facilityAnalystSetting = getObjFromList(facilityAnalystSettingId);
            facilityAnalystSetting.setTNodeIDField(value);

            promise.resolve(true);
        }catch (Exception e){
            promise.reject(e);
        }
    }

    /**
     * 设置点到弧段的距离容限
     * @param facilityAnalystSettingId
     * @param value
     * @param promise
     */
    @ReactMethod
    public void setTolerance(String facilityAnalystSettingId, double value, Promise promise) {
        try{
            FacilityAnalystSetting facilityAnalystSetting = getObjFromList(facilityAnalystSettingId);
            facilityAnalystSetting.setTolerance(value);

            promise.resolve(true);
        }catch (Exception e){
            promise.reject(e);
        }
    }

    /**
     * 设置权值字段信息集合对象。
     * @param facilityAnalystSettingId
     * @param weightFieldInfosId
     * @param promise
     */
    @ReactMethod
    public void setWeightFieldInfos(String facilityAnalystSettingId, String weightFieldInfosId, Promise promise) {
        try{
            FacilityAnalystSetting facilityAnalystSetting = getObjFromList(facilityAnalystSettingId);

            WeightFieldInfos weightFieldInfos = JSWeightFieldInfos.getObjFromList(weightFieldInfosId);

            facilityAnalystSetting.setWeightFieldInfos(weightFieldInfos);

            promise.resolve(true);
        }catch (Exception e){
            promise.reject(e);
        }
    }

}

File: ios/objNative/JSFacilityAnalystSetting.m
//
//  JSFacilityAnalystSetting.m
//  Supermap
//
//  Created by supermap on 2018/8/8.
//  Copyright © 2018年 Facebook. All rights reserved.
//

#import "JSFacilityAnalystSetting.h"
#import "JSObjManager.h"
#import "SuperMap/FacilityAnalystSetting.h"
@interface JSFacilityAnalystSetting(){
    FacilityAnalystSetting* m_FacilityAnalystSetting;
}
@end
@implementation JSFacilityAnalystSetting
RCT_EXPORT_MODULE();

RCT_REMAP_METHOD(createObj, createObjResolver:(RCTPromiseResolveBlock)resolve rejecter:(RCTPromiseRejectBlock)reject){
    @try {
        FacilityAnalystSetting* facilityAnalystSetting = [[FacilityAnalystSetting alloc]init];
        NSInteger facilityAnalystSettingId = (NSInteger)facilityAnalystSetting;
        [JSObjManager addObj:facilityAnalystSetting];
        resolve(@(facilityAnalystSettingId).stringValue);    
    } @catch (NSException *exception) {
        reject(@"JSFacilityAnalystSetting",@"createObj expection",nil);
    }
}
/**
 * 返回障碍弧段的 ID 列表
 * @param facilityAnalystSettingId
 */
RCT_REMAP_METHOD(getBarrierEdges,getBarrierEdgesById:(NSString*)facilityAnalystSettingId resolver:(RCTPromiseResolveBlock)resolve rejecter:(RCTPromiseRejectBlock)reject){
    @try {
        m_FacilityAnalystSetting = [JSObjManager getObjWithKey:facilityAnalystSettingId];
        NSMutableArray * ids = m_FacilityAnalystSetting.barrierEdges;
        resolve(ids);
    } @catch (NSException *exception) {
        reject(@"JSFacilityAnalystSetting",@"getBarrierEdges expection",nil);
    }
}
/**
 * 返回障碍结点的 ID 列表
 * @param facilityAnalystSettingId
 */
RCT_REMAP_METHOD(getBarrierNodes,getBarrierNodesById:(NSString*)facilityAnalystSettingId resolver:(RCTPromiseResolveBlock)resolve rejecter:(RCTPromiseRejectBlock)reject){
    @try {
        m_FacilityAnalystSetting = [JSObjManager getObjWithKey:facilityAnalystSettingId];
        NSMutableArray * ids = m_FacilityAnalystSetting.barrierNodes;
        resolve(ids);
    } @catch (NSException *exception) {
        reject(@"JSFacilityAnalystSetting",@"getBarrierNodes expection",nil);
    }
}
/**
 * 返回流向字段
 * @param facilityAnalystSettingId
 * @param promise
 */
RCT_REMAP_METHOD(getDirectionField,getDirectionFieldById:(NSString*)facilityAnalystSettingId resolver:(RCTPromiseResolveBlock)resolve rejecter:(RCTPromiseRejectBlock)reject){
    @try {
        m_FacilityAnalystSetting = [JSObjManager getObjWithKey:facilityAnalystSettingId];
        NSString* directionField = m_FacilityAnalystSetting.directionField;
        resolve(directionField);
    } @catch (NSException *exception) {
        reject(@"JSFacilityAnalystSetting",@"getDirectionField expection",nil);
    }
}
/**
 * 返回网络数据集中标识弧段 ID 的字段
 * @param facilityAnalystSettingId
 */
RCT_REMAP_METHOD(getEdgeIDField,getEdgeIDFieldById:(NSString*)facilityAnalystSettingId resolver:(RCTPromiseResolveBlock)resolve rejecter:(RCTPromiseRejectBlock)reject){
    @try {
        m_FacilityAnalystSetting = [JSObjManager getObjWithKey:facilityAnalystSettingId];
        NSString* edgeIDField = m_FacilityAnalystSetting.edgeIDField;
        resolve(edgeIDField);
    } @catch (NSException *exception) {
        reject(@"JSFacilityAnalystSetting",@"getEdgeIDField expection",nil);
    }
}
/**
 * 返回网络数据集中标识弧段起始结点 ID 的字段
 * @param facilityAnalystSettingId
 * @param promise
 */
RCT_REMAP_METHOD(getFNodeIDField,getFNodeIDFieldById:(NSString*)facilityAnalystSettingId resolver:(RCTPromiseResolveBlock)resolve rejecter:(RCTPromiseRejectBlock)reject){
    @try {
        m_FacilityAnalystSetting = [JSObjManager getObjWithKey:facilityAnalystSettingId];
        NSString* fNodeIDField = m_FacilityAnalystSetting.fNodeIDField;
        resolve(fNodeIDField);
    } @catch (NSException *exception) {
        reject(@"JSFacilityAnalystSetting",@"getFNodeIDField expection",nil);
    }
}
/**
 * 返回网络数据集
 * @param facilityAnalystSettingId
 */
RCT_REMAP_METHOD(getNetworkDataset,getNetworkDatasetById:(NSString*)facilityAnalystSettingId resolver:(RCTPromiseResolveBlock)resolve rejecter:(RCTPromiseRejectBlock)reject){
    @try {
        m_FacilityAnalystSetting = [JSObjManager getObjWithKey:facilityAnalystSettingId];
        DatasetVector* datasetVector = m_FacilityAnalystSetting.netWorkDataset;
        NSInteger datasetVectorId = (NSInteger)datasetVector;
        [JSObjManager addObj:datasetVector];
        resolve(@(datasetVectorId).stringValue);
    } @catch (NSException *exception) {
        reject(@"JSFacilityAnalystSetting",@"getNetworkDataset expection",nil);
    }
}
/**
 * 返回网络数据集中标识网络结点 ID 的字段
 * @param facilityAnalystSettingId
 */
RCT_REMAP_METHOD(getNodeIDField,getNodeIDFieldById:(NSString*)facilityAnalystSettingId resolver:(RCTPromiseResolveBlock)resolve rejecter:(RCTPromiseRejectBlock)reject){
    @try {
        m_FacilityAnalystSetting = [JSObjManager getObjWithKey:facilityAnalystSettingId];
        NSString* nodeIDField = m_FacilityAnalystSetting.nodeIDField;
        resolve(nodeIDField);
    } @catch (NSException *exception) {
        reject(@"JSFacilityAnalystSetting",@"getNodeIDField expection",nil);
    }
}
/**
 * 返回网络数据集中标识弧段终止结点 ID 的字段
 * @param facilityAnalystSettingId
 */
RCT_REMAP_METHOD(getTNodeIDField,getTNodeIDFieldById:(NSString*)facilityAnalystSettingId resolver:(RCTPromiseResolveBlock)resolve rejecter:(RCTPromiseRejectBlock)reject){
    @try {
        m_FacilityAnalystSetting = [JSObjManager getObjWithKey:facilityAnalystSettingId];
        NSString* tNodeIDField = m_FacilityAnalystSetting.tNodeIDField;
        resolve(tNodeIDField);
    } @catch (NSException *exception) {
        reject(@"JSFacilityAnalystSetting",@"getTNodeIDField expection",nil);
    }
}
/**
 * 返回点到弧段的距离容限
 * @param facilityAnalystSettingId
 */
RCT_REMAP_METHOD(getTolerance,getToleranceById:(NSString*)facilityAnalystSettingId resolver:(RCTPromiseResolveBlock)resolve rejecter:(RCTPromiseRejectBlock)reject){
    @try {
        m_FacilityAnalystSetting = [JSObjManager getObjWithKey:facilityAnalystSettingId];
        double tolerance = m_FacilityAnalystSetting.tolerance;
        resolve(@(tolerance));
    } @catch (NSException *exception) {
        reject(@"JSFacilityAnalystSetting",@"getTolerance expection",nil);
    }
}
/**
 * 返回权值字段信息集合对象
 * @param facilityAnalystSettingId
 */
RCT_REMAP_METHOD(getWeightFieldInfos,getWeightFieldInfosById:(NSString*)facilityAnalystSettingId resolver:(RCTPromiseResolveBlock)resolve rejecter:(RCTPromiseRejectBlock)reject){
    @try {
        m_FacilityAnalystSetting = [JSObjManager getObjWithKey:facilityAnalystSettingId];
        WeightFieldInfos* weightFieldInfos = m_FacilityAnalystSetting.weightFieldInfos;
        NSInteger weightFieldInfosId= (NSInteger)weightFieldInfos;
        [JSObjManager addObj:weightFieldInfos];
        resolve(@(weightFieldInfosId).stringValue);
    } @catch (NSException *exception) {
        reject(@"JSFacilityAnalystSetting",@"getWeightFieldInfos expection",nil);
    }
}
/**
 * 设置障碍弧段的 ID 列表
 * @param facilityAnalystSettingId
 * @param value
 */
RCT_REMAP_METHOD(setBarrierEdges,setBarrierEdgesById:(NSString*)facilityAnalystSettingId value:(NSMutableArray*)value resolver:(RCTPromiseResolveBlock)resolve rejecter:(RCTPromiseRejectBlock)reject){
    @try {
        m_FacilityAnalystSetting = [JSObjManager getObjWithKey:facilityAnalystSettingId];
        m_FacilityAnalystSetting.barrierEdges =value;
        NSNumber* number = [NSNumber numberWithBool:YES];
        resolve(number);
    } @catch (NSException *exception) {
        reject(@"JSFacilityAnalystSetting",@"setBarrierEdges expection",nil);
    }
}
/**
 * 设置障碍结点的 ID 列表
 * @param facilityAnalystSettingId
 * @param value
 */
RCT_REMAP_METHOD(setBarrierNodes,setBarrierNodesById:(NSString*)facilityAnalystSettingId value:(NSMutableArray*)value resolver:(RCTPromiseResolveBlock)resolve rejecter:(RCTPromiseRejectBlock)reject){
    @try {
        m_FacilityAnalystSetting = [JSObjManager getObjWithKey:facilityAnalystSettingId];
        m_FacilityAnalystSetting.barrierNodes =value;
        NSNumber* number = [NSNumber numberWithBool:YES];
        resolve(number);
    } @catch (NSException *exception) {
        reject(@"JSFacilityAnalystSetting",@"checksetBarrierNodesLoops expection",nil);
    }
}
/**
 * 设置流向字段
 * @param facilityAnalystSettingId
 * @param value
 */
RCT_REMAP_METHOD(setDirectionField,setDirectionFieldById:(NSString*)facilityAnalystSettingId value:(NSString*)value resolver:(RCTPromiseResolveBlock)resolve rejecter:(RCTPromiseRejectBlock)reject){
    @try {
        m_FacilityAnalystSetting = [JSObjManager getObjWithKey:facilityAnalystSettingId];
        m_FacilityAnalystSetting.directionField =value;
        NSNumber* number = [NSNumber numberWithBool:YES];
        resolve(number);
    } @catch (NSException *exception) {
        reject(@"JSFacilityAnalystSetting",@"setDirectionField expection",nil);
    }
}
/**
 * 设置网络数据集中标识弧段 ID 的字段
 * @param facilityAnalystSettingId
 * @param value
 */
RCT_REMAP_METHOD(setEdgeIDField,setEdgeIDFieldById:(NSString*)facilityAnalystSettingId value:(NSString*)value resolver:(RCTPromiseResolveBlock)resolve rejecter:(RCTPromiseRejectBlock)reject){
    @try {
        m_FacilityAnalystSetting = [JSObjManager getObjWithKey:facilityAnalystSettingId];
        m_FacilityAnalystSetting.edgeIDField =value;
        NSNumber* number = [NSNumber numberWithBool:YES];
        resolve(number);
    } @catch (NSException *exception) {
        reject(@"JSFacilityAnalystSetting",@"setEdgeIDField expection",nil);
    }
}
/**
 * 设置网络数据集中标识弧段起始结点 ID 的字段
 * @param facilityAnalystSettingId
 */
RCT_REMAP_METHOD(setFNodeIDField,setFNodeIDFieldById:(NSString*)facilityAnalystSettingId value:(NSString*)value resolver:(RCTPromiseResolveBlock)resolve rejecter:(RCTPromiseRejectBlock)reject){
    @try {
        m_FacilityAnalystSetting = [JSObjManager getObjWithKey:facilityAnalystSettingId];
        m_FacilityAnalystSetting.fNodeIDField =value;
        NSNumber* number = [NSNumber numberWithBool:YES];
        resolve(number);
    } @catch (NSException *exception) {
        reject(@"JSFacilityAnalystSetting",@"setFNodeIDField expection",nil);
    }
}
/**
 * 设置网络数据集
 * @param facilityAnalystSettingId
 * @param datasetVectorId
 */
RCT_REMAP_METHOD(setNetworkDataset,setNetworkDatasetById:(NSString*)facilityAnalystSettingId datasetVectorId:(NSString*)datasetVectorId resolver:(RCTPromiseResolveBlock)resolve rejecter:(RCTPromiseRejectBlock)reject){
    @try {
        m_FacilityAnalystSetting = [JSObjManager getObjWithKey:facilityAnalystSettingId];
        DatasetVector* datasetVector = [JSObjManager getObjWithKey:datasetVectorId];
        m_FacilityAnalystSetting.netWorkDataset=datasetVector;
        NSNumber* number = [NSNumber numberWithBool:YES];
        resolve(number);
    } @catch (NSException *exception) {
        reject(@"JSFacilityAnalystSetting",@"setNetworkDataset expection",nil);
    }
}
/**
 * 设置网络数据集中标识网络结点 ID 的字段
 * @param facilityAnalystSettingId
 * @param value
 */
RCT_REMAP_METHOD(setNodeIDField,setNodeIDFieldById:(NSString*)facilityAnalystSettingId value:(NSString*)value resolver:(RCTPromiseResolveBlock)resolve rejecter:(RCTPromiseRejectBlock)reject){
    @try {
        m_FacilityAnalystSetting = [JSObjManager getObjWithKey:facilityAnalystSettingId];
        m_FacilityAnalystSetting.nodeIDField =value;
        NSNumber* number = [NSNumber numberWithBool:YES];
        resolve(number);
    } @catch (NSException *exception) {
        reject(@"JSFacilityAnalystSetting",@"setNodeIDField expection",nil);
    }
}

/**
 * 设置网络数据集中标识弧段终止结点 ID 的字段
 * @param facilityAnalystSettingId
 * @param value
 */
RCT_REMAP_METHOD(setTNodeIDField,setTNodeIDFieldById:(NSString*)facilityAnalystSettingId value:(NSString*)value resolver:(RCTPromiseResolveBlock)resolve rejecter:(RCTPromiseRejectBlock)reject){
    @try {
        m_FacilityAnalystSetting = [JSObjManager getObjWithKey:facilityAnalystSettingId];
        m_FacilityAnalystSetting.tNodeIDField =value;
        NSNumber* number = [NSNumber numberWithBool:YES];
        resolve(number);
    } @catch (NSException *exception) {
        reject(@"JSFacilityAnalystSetting",@"setTNodeIDField expection",nil);
    }
}
/**
 * 设置点到弧段的距离容限
 * @param facilityAnalystSettingId
 * @param value
 */
RCT_REMAP_METHOD(setTolerance,setToleranceById:(NSString*)facilityAnalystSettingId value:(double)value resolver:(RCTPromiseResolveBlock)resolve rejecter:(RCTPromiseRejectBlock)reject){
    @try {
        m_FacilityAnalystSetting = [JSObjManager getObjWithKey:facilityAnalystSettingId];
        m_FacilityAnalystSetting.tolerance =value;
        NSNumber* number = [NSNumber numberWithBool:YES];
        resolve(number);
    } @catch (NSException *exception) {
        reject(@"JSFacilityAnalystSetting",@"setTolerance expection",nil);
    }
}
/**
 * 设置权值字段信息集合对象。
 * @param facilityAnalystSettingId
 * @param weightFieldInfosId
 */
RCT_REMAP_METHOD(setWeightFieldInfos,setWeightFieldInfosById:(NSString*)facilityAnalystSettingId weightFieldInfosId:(NSString*)weightFieldInfosId resolver:(RCTPromiseResolveBlock)resolve rejecter:(RCTPromiseRejectBlock)reject){
    @try {
        m_FacilityAnalystSetting = [JSObjManager getObjWithKey:facilityAnalystSettingId];
        WeightFieldInfos* weightFieldInfos =[JSObjManager getObjWithKey:weightFieldInfosId];
        m_FacilityAnalystSetting.weightFieldInfos = weightFieldInfos;
        NSNumber* number = [NSNumber numberWithBool:YES];
        resolve(number);
    } @catch (NSException *exception) {
        reject(@"JSFacilityAnalystSetting",@"setWeightFieldInfos expection",nil);
    }
}
@end
